ambry-api/src/main/java/com/github/ambry/config/NetworkConfig.java

```diff
@@ -170,6 +177,8 @@ public NetworkConfig(VerifiableProperties verifiableProperties) {
     requestQueueType = verifiableProperties.getEnum(REQUEST_QUEUE_TYPE, RequestQueueType.class,
         RequestQueueType.BASIC_QUEUE_WITH_FIFO);
     requestQueueTimeoutMs = verifiableProperties.getIntInRange(REQUEST_QUEUE_TIMEOUT_MS, 2000, 0, Integer.MAX_VALUE);
+    // Used 5000 since Ambry incoming max qps seems to be 2.5K in last few weeks and default request.queue.timeout is 2 seconds
+    requestQueueCapacity = verifiableProperties.getIntInRange(REQUEST_QUEUE_CAPACITY, 5000, 0, Integer.MAX_VALUE);
     adaptiveLifoQueueCodelTargetDelayMs =
```

Comment:
1. Does queue have requests or chunks ? If requests, then can the requests be of arbitrary size ?
2. 20GB is ok, that's where most servers are but I'd say make it even more like 25,000 if each item in queue is no more than 4MB ? That way the outliers will stick out. Otherwise we cannot distinguish a bad host from a good one because all are 25-30g directmem. 100g of direct-mem will not crash server because the upper limit is 150gb.